File: submissions/57335b62a6200f1877712191/src/SRD.Words/SRD.Words/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SRD.Words
{
    public class Node<T>
    {
        private IList<Node<T>> _children;

        public Node(Node<T> parent, T value)
        {
            Parent = parent;
            Value = value;
            _children = new List<Node<T>>();
        }

        public Node<T> Parent { get; set; }
        public Node<T>[] Children
        {
            get
            {
                return _children.ToArray();
            }
        }

        public T Value { get; private set; }
        public bool IsLeaf { get; set; }

        public void AddChild(Node<T> child)
        {
            _children.Add(child);
        }

        public void RemoveChild(Node<T> node)
        {
            if(_children.Contains(node))
            {
                _children.Remove(node);
            }
        }

        public WordPartStatistics Tag { get; set; }
    }
}

File: submissions/5748a99163905b3a11d97ca7/src/node.js
module.exports = function node() {

    this.parent = null;
    this.children = [];
    this.value = null;

    this.setParent = function (parent) {
        this.parent = parent;
    };

    this.getParent = function(){
        return this.parent;
    };

    this.addChild = function(child){
        this.children.push(child);
    };

    this.getChildren = function(){
        return this.children;
    };

    this.setValue = function(oValue){
        this.value = oValue;
    };

    this.getValue = function(){
        return this.value;
    };


};

